Src/Newtonsoft.Json/Linq/JConstructor.cs

```diff
@@ -164,9 +164,10 @@ public override void WriteTo(JsonWriter writer, params JsonConverter[] converter
         {
             writer.WriteStartConstructor(_name);
 
-            foreach (JToken token in Children())
+            int count = _values.Count;
```

Comment:
I tend to go for one extreme or the other; either just use `foreach` or do every little tweak available. In any case though, done as requested.